### PR TITLE
Rename ToolActions to ItemAbilities

### LINKED ISSUE #1032: Rename ToolActions in 1.21
This came up as a side tangent to some conversations but basically ToolActions are like Tags but nbt sensitive. Making them useful for more than just tag actions. So perhaps a rename would be on the table?

Shadow suggested ItemAbility.

Any other names or opposition to rename?

### PR DESCRIPTION
**Pull request by TelepathicGrunt**
The goal is that ToolActions looks like it is only for tools that do some sort of actions. But that is not what this API actually is. The API is basically a marker to signal your item for any sort of itemstack sensitive check easily. Even between mods without any compile dependency. 

After asking people for some names, ItemAbilities came up multiple times independently. This means this name is the most intuitive one that people keep thinking of when describing ToolActions to them. 

Closes https://github.com/neoforged/NeoForge/issues/1032

### CODE REVIEW

**Review by pupnewfster — DISMISSED** [review]

File: patches/net/minecraft/world/level/block/TripWireBlock.java.patch [hunk 1]
```diff
@@ -1,11 +1,19 @@
 --- a/net/minecraft/world/level/block/TripWireBlock.java
 +++ b/net/minecraft/world/level/block/TripWireBlock.java
+@@ -3,6 +_,7 @@
+ import com.mojang.serialization.MapCodec;
+ import com.mojang.serialization.codecs.RecordCodecBuilder;
+ import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
++
```
Comment (ThatGravyBoat) [inline patches/net/minecraft/world/level/block/TripWireBlock.java.patch]: it seems the imports were unintentionally touched here.

Comment (TelepathicGrunt) [reply patches/net/minecraft/world/level/block/TripWireBlock.java.patch]: yeah that was when I did IDE rename on ToolActions... big mistake. Intellij optimized imports and I had to go fix them by hand.

**Review by pupnewfster — DISMISSED** [review]

**Comment by TelepathicGrunt** [comment]
Yo neo modders, do we want to add namespaces to these ItemAbilities (ToolActions) strings? So `shears_dig` becomes `neoforge:shears_dig`. Speak now

**Comment by RaymondBlaze** [comment]
> Yo neo modders, do we want to add namespaces to these ItemAbilities (ToolActions) strings? So `shears_dig` becomes `neoforge:shears_dig`. Speak now

Then it also applies to the `EffectCure` system, since they share some level of similarity where the instances are computed if absent and represented from `String` inputs.
I personally agree with switching to `ResourceLocation` for that, since all the benefits for using a `String` other than a `ResourceLocation` could be covered by using the conventional namespace `c`, and switching to `ResourceLocation` will make them be more consistent with other stuffs.

**Comment by TelepathicGrunt** [comment]
Me and the other maintainers talked some more and decided it may be best to not namespace. The existing ItemAbilities in Neo would’ve gotten the neo namespace which would make modders not realize they should use c or some other namespace for common ItemAbilities. So it’ll lead to most modders using neo namespace and at that point, there’s just no need for namespaces. We aren’t seeing much benefit to namespacing that outweighs the confusion over what namespace to use when

Convo: https://discord.com/channels/313125603924639766/1154167065519861831/1254934108610756740

(Note, maintainer's decisions could change in more discussions back to namespaces. May still be on-going)

**Review by pupnewfster — DISMISSED** [review]

**Review by pupnewfster — APPROVED** [review]

**Pull request merged**